packages/dev/core/src/Meshes/Compression/dracoCodec.ts

```diff
@@ -0,0 +1,206 @@
+import type { Nullable } from "core/types";
+import { Tools } from "../../Misc/tools";
+import { AutoReleaseWorkerPool } from "../../Misc/workerPool";
+import type { IDisposable } from "../../scene";
+import { initializeWebWorker } from "./dracoCompressionWorker";
+
+/**
+ * Configuration for using a Draco codec.
+ */
+export interface IDracoCodecConfiguration {
+    /**
+     * The url to the WebAssembly module.
+     */
+    wasmUrl?: string;
+
+    /**
+     * The url to the WebAssembly binary.
+     */
+    wasmBinaryUrl?: string;
+
+    /**
+     * The url to the fallback JavaScript module.
+     */
+    fallbackUrl?: string;
+
+    /**
+     * The number of workers for async operations. Specify `0` to disable web workers and run synchronously in the current context.
+     */
+    numWorkers?: number;
+
+    /**
+     * Optional worker pool to use for async encoding/decoding.
+     * If provided, numWorkers will be ignored and the worker pool will be used instead.
+     * If provided the draco script will not be loaded from the DracoConfiguration.
+     */
+    workerPool?: AutoReleaseWorkerPool;
+
+    /**
+     * Optional ArrayBuffer of the WebAssembly binary.
+     * If provided it will be used instead of loading the binary from wasmBinaryUrl.
+     */
+    wasmBinary?: ArrayBuffer;
+
+    /**
+     * The codec module if already available.
+     */
+    jsModule?: any /* DracoDecoderModule | DracoEncoderModule */;
+}
+
+/**
+ * @internal
+ */
+export function _GetDefaultNumWorkers(): number {
+    if (typeof navigator !== "object" || !navigator.hardwareConcurrency) {
+        return 1;
+    }
+
+    // Use 50% of the available logical processors but capped at 4.
+    return Math.min(Math.floor(navigator.hardwareConcurrency * 0.5), 4);
+}
+
+/**
+ * Base class for a Draco codec.
+ * @internal
+ */
+export abstract class DracoCodec<M> implements IDisposable {
+    protected _workerPoolPromise?: Promise<AutoReleaseWorkerPool>;
+    protected _modulePromise?: Promise<{ module: M }>;
+
+    /**
+     * The configuration for the Draco codec.
+     * Subclasses should override this value with a valid configuration.
+     */
+    public static Config: IDracoCodecConfiguration;
+
+    /**
+     * Returns true if the codec's `Configuration` is available.
+     */
+    public static get Available(): boolean {
+        return !!((this.Config.wasmUrl && this.Config.wasmBinaryUrl && typeof WebAssembly === "object") || this.Config.fallbackUrl);
+    }
+
+    /**
+     * The default draco compression object
+     * Subclasses should override this value using the narrowed type of the subclass,

```

Comment:
I am not sure this will work as you expect. As there is no `this` when using static you will need to use the actual static member of a specific class. I can explain it more specifically, but there should probably be a different approach here.
Having said that, it would be better, in general, to avoid static as much as possible, and move them to the module, wherever possible (like you did to `_GetDefaultNumWorkers`)

Reply:
Talked about this with you earlier, so just going to leave a note here: 
Using `this` to access derived classes' statics is not good practice, but it works for this case.

I'm going to wait for any other thoughts in the unresolved conversation above about the Config before updating this.